src/pools/utils.ts

```diff
@@ -173,9 +173,9 @@ const _getUserClaimable = async (pools: string[], address: string, useCache: boo
             }
 
             for (const token of rewardTokens[poolId]) {
-                _setContracts(token, ERC20Abi);
-                const tokenMulticallContract = curve.contracts[token].multicallContract;
-                rewardInfoCalls.push(tokenMulticallContract.symbol(), tokenMulticallContract.decimals());
+                // Don't reset the reward ABI if the reward is the LP token itself, otherwise we lose LP contract functions
+                const { multicallContract } = token !== pool.address ? _setContracts(token, ERC20Abi) : curve.contracts[token]
```

Comment:
Double negatives are hard to read
```suggestion
                const { multicallContract } = token === pool.address ? curve.contracts[token] : _setContracts(token, ERC20Abi)
```

Reply:
Would it make sense to move this to _setContracts?

Reply:
I thought about it but decided not to, because it's a pure setter function. In my view it would be confusing if a setter was called and nothing would be set